.github/gh-actions-self-hosted-runners/helper-functions/cloud-functions/monitorRunnersStatus/index.js

```diff
@@ -0,0 +1,86 @@
+//  Licensed to the Apache Software Foundation (ASF) under one
+//  or more contributor license agreements.  See the NOTICE file
+//  distributed with this work for additional information
+//  regarding copyright ownership.  The ASF licenses this file
+//  to you under the Apache License, Version 2.0 (the
+//  "License"); you may not use this file except in compliance
+//  with the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing,
+//  software distributed under the License is distributed on an
+//  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+//  KIND, either express or implied.  See the License for the
+//  specific language governing permissions and limitations
+//  under the License.
+
+//This function will return the number of online and offline runners for
+// each OS (Windows, linux), an additional Github actions workflow will run
+// to request this Cloud Function and send alerts based on the status.
+
+import functions from '@google-cloud/functions-framework';
+import { Octokit } from "octokit";
+import { createAppAuth } from "@octokit/auth-app";
+
+async function monitorRunnerStatus() {
+    try {
+        //Set your GH App values as environment variables
+        let authOptions = {
+            appId: process.env.APP_ID,
+            privateKey: process.env.PEM_KEY,
+            clientId: process.env.CLIENT_ID,
+            clientSecret: process.env.CLIENT_NAME,
+            installationId: process.env.APP_INSTALLATION_ID
+        }
+        const octokit = new Octokit({
+            authStrategy: createAppAuth,
+            auth: authOptions
+        });
+        let runners = await octokit.request(`GET /orgs/${process.env.ORG}/actions/runners`, {
+            org: process.env.ORG,
```

Comment:
Should we add some validation that `process.env.ORG` is set when this file is executed?

Reply:
Yes, we have now included an env validation block, could you help us with an additional review for this and the other modifications? Thanks in advance.